### PR TITLE
Allow permanent redirects to be used

### PR DESCRIPTION
**Pull request by TiddoLangerak**
Permanent redirects can result in a significant speed up, since they
don't require to be re-fetched. With a permanent redirect images can
thus be served directly from cache. On pages with many images - where
concurrent requests are throttled - this can result in a significant
speed up.

The use of the permanent redirect is configurable: if a non-zero
redirect timeout is set then a permanent redirect is used, otherwise it
still uses a temporary one. This therefore doesn't change anything for
existing configurations, it needs to be explicitly enabled.

@rogierslag - I wanted to test this, but I don't have any AWS keys available. Do we have some for testing purposes?

### CODE REVIEW

File: config/default.json.example [hunk 1]
```diff
@@ -13,13 +13,14 @@
...
   "allow_indexing": false,
   "constraints": {
     "max_width": 2000,
     "max_height": 2000
   },
   "log": {
     "debug": false
-  }
+  },
+  "redirect_timeout" : 604800
```
Comment (rogierslag) [inline config/default.json.example]: Add the parameter description to the readme as well (I think the others are already in there)

**Comment by rogierslag** [comment]
You can obtain Magnet.me credentials through the AWS dashboard if required for testing.

**Comment by TiddoLangerak** [comment]
Couldn't find the settings in the `README`, so I added all that I could find in the example config. You might want to double check if I got them right though. 

I also made a small but important change: Somewhere in our long discussion about this I think you mentioned problems we had earlier with non-standard complaint implementations of permanent redirects, that cache redirects for way longer than they're supposed to. I think you're right on that, and on second thought I also think that semantically the permanent redirect isn't the right solution because it (semantically) changes the representative URL of the resource, which is generally undesirable. I therefore changed it back to only use temporary redirects, but with cache control headers. For standard complaint implementations this should have the same net results (and also the same caching concerns), but it is likely to give less troubles in edge cases. 

I still need to test it, so I added the DNM label for it.

**Comment by rogierslag** [comment]
Note that many browsers do not cache 307 responses. We had this before, but removed it since it was useless https://github.com/inventid/iaas/pull/77#discussion-diff-69939479

**Comment by TiddoLangerak** [comment]
That's a bummer... Then I think 301 is the way to go after all, it's a bit useless otherwise.

**Comment by rogierslag** [comment]
The 307 RFC does suggest that it is cachable, but we found some browsers which ignore it anyway 

https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html

**Comment by rogierslag** [comment]
Tested it, and it seems the 307 is properly cached now (by at least Chrome). So I'd sugest to give this a try, and let's see how it performs in the real world :)

Made a small adjustment: added a shrinkwrap since something changed in Babel which causes the Docker container to be unable to build. So I shrinkwrapped my local build, which does work

**Comment by rogierslag** [comment]
So feel free to merge if you agree :)
